### DISCUSSION TITLE
Unzip-only install of (Win11) WT, crashing in Win 11

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#14163] [Q&A] [answered]
Unzip-only installs of the Win10 distrib of WT work OK in Win11.

With an unzip-only install of the Win11 distrib, WT crashes on startup.  It that expected.  I have dumps.

**Comment by DHowett** [comment, marked as answer]
Thanks for trying it out! That is expected, and documented. :smile:

<img width="666" alt="image" src="https://user-images.githubusercontent.com/189190/194663720-4ca1db67-a42b-4719-8c02-157dccdbe2c2.png">

The "Windows 10" build is just "A build of Windows Terminal with no package dependencies." It contains everything it needs to run standalone.

**Reply by vefatica** [reply]
What a fiasco!  Here are 3 hours worth of problems, in order of increasing severity.

My homemede permanent drive SUBSTs (DOS Devices) were gone ... expected ... I could have re-created them.

Couldn't get volume control out of the notification area.  Googling suggested it's impossible.

While I could print TXT files and a test page, neither Firefox nor Adobe could print a PDF file.  And the printer troubleshooter would not run.  I don't know if I could have fixed those.

And (the clincher) WinWord couldn't open a rather important password-protected DOCX file because it was "locked by another user" (?).  That remained true after many restarts.  I checked the file's security ... only the owner, jj\vefatica (with full access), mentioned there.

All I wanted was a newer ConHost.exe that would be updated.  Oh well!  I'm back to Windows 10 and the items above are no longer problematic.  At least the roll-back part of the update worked well.  :-)

**Reply by DHowett** [reply]
Sheesh! I'm sorry you had to go through all that. Thank you for trying, nonetheless.

I'll let you in on a bit of a secret: If you replace `conhost.exe` with `openconsole.exe` from the Windows Terminal distribution, you'll get all the new features and bug fixes. Sure, it's under system file protection, but... I'm sure you've got a way around that.

There are only three things that are missing from OpenConsole that are present in conhost:

1. The dark title bar (ugh)
2. Enterprise Data Protection for the clipboard (pasting EDP-protected data into OpenConsole will pop up a "no, you may not pass Go" dialog)
3. Permission enforcement on the text buffer for UWP console applications (since there appear to be only two of these in the wild, it is almost a guarantee that this will not impact you; also, since permissions are enforced in stock conhost they will not have been authored to try to break them :P)

**Reply by vefatica** [reply]
I've tried it and your #1 is why I don't leave it that way.  Ain't gonna happen, eh?

**Reply by vefatica** [reply]
If starting a shell with the likes of  `d:\wtp\exe\OpenConsole.exe d:\tc29\tcc.exe` is an appropriate test (I'm not 100% sure of that), then there's another difference.  I have an app to switch to any of WT's 9 themes.  It's ultimately does this (below).  Note the comment.

```
WriteConsoleA(hConOut, szThemeSequence, nSeqLen, nullptr, nullptr);
// clear whole buffer; apparently necessary in Win10's console, but not in Windows Terminal or OpenConsole.exe
WriteConsoleA(hConOut, "\x1b[2J\x1b[3J\x1b[H", 11, nullptr, nullptr);
```

**Reply by DHowett** [reply]
I believe that that was a bug that's been fixed since Windows 10 :smile:

Say, do you care about the _titlebar and the scrollbar_ being dark, or just the titlebar?

**Reply by vefatica** [reply]
I get the same color title bar whether I run `cmd` (win10 conhost) or `d:\wtp\exe\OpenConsole.exe cmd`, namely my chosen accent color.  It's the white scrollbar I don't like.

**Reply by DHowett** [reply]
Alas, that's the one we can't fix out in the open. Sorry :/

(I'll apply some pressure...)

**Reply by vefatica** [reply]
Obviously I didn't read your #1 carefully.  I read it as mentioning the scrollbar.  As I said, title bar colors are OK.  FWIW, these are what I see (openconsole on the top, conhost on the bottom); the second has focus reversed.

![image](https://user-images.githubusercontent.com/61856645/194938878-1e102e0e-93eb-4a69-87fb-496f69559e1a.png)

![image](https://user-images.githubusercontent.com/61856645/194938953-2cc7fd2d-b1be-4a73-a0ef-dc94b03a471f.png)

**Reply by vefatica** [reply]
"Out in the open" ... huh?

**Reply by DHowett** [reply]
Ah, as in "using code that I can share with our community." Using the appropriate dark control theme is one of the three things that are built closed-source into conhost.
